### PR TITLE
Add helper cmd-line program for just combining files

### PR DESCRIPTION
**Pull request by jlmaurer**
Running ```raiderCombine.py --raiderDir '*/' --raider all_delays.csv``` combines all the delay files together (in this case, without matching to a GNSS delay file)

## Description

- Filenames still need to have the datetime string "YYYYMMDDTHHMMSS" in their name, which all RAiDER-generated files do. 
- Updated the search for the datetime in the filename to be a regex instead of integer indexing
- Made the GNSS file optional so that you can combine RAiDER files without having a GNSS file
- Any set of files you can find by globbing are discoverable
- The output RAiDER delay file is sorted by first station ID then by Datetime

- all unit tests pass locally
- Added unit tests to test/test_gnss.py

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have added an explanation of what your changes do and why you'd like us to include them.
- [x] I have written new tests for your core changes, as applicable.
- [x] I have successfully ran tests with your changes locally.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by jlmaurer** [comment]
@dbekaert and @sssangha this is a slightly modified version of the code that allows for combing RAiDER delay files without needing to have a GNSS file for comparison. You can simply omit the --gnss option and combine the files. Also note that it should work as expected regardless of datetime / region; i.e. it will take what is there and concatenate, and I was already dropping dups so did not need to add that.

**Comment by jlmaurer** [comment]
Just realized that this is still taking the inner merge. Just as a reminder: 
- Inner only keep datetime/ID combos that have both GNSS and RAiDER delays
- left keeps (in this case) all raider delays and matches up GNSS delays as available
So we want to switch to left merge?

**Comment by dbekaert** [comment]
@jlmaurer; yes left merge is probably the way to go. But would be good to report out a stat per year on how much of the GNSS is missing.

File: tools/RAiDER/gnss/processDelayFiles.py [hunk 1]
```diff
@@ -143,12 +162,20 @@ def create_parser():
...
     )
 
+    p.add_argument(
+        '--gnss', dest='gnss_file',
+        help=dedent("""\
+            Optional .csv file containing GPS Zenith Delays. Should contain columns "ID", "ZTD", and "Datetime"
+            """),
+        default=None
+    )
+
```
Comment (dbekaert) [inline tools/RAiDER/gnss/processDelayFiles.py:172]: Can you add run file examples to the helper?

Comment (jlmaurer) [reply tools/RAiDER/gnss/processDelayFiles.py:173]: @dbekaert done.

Comment (jlmaurer) [reply tools/RAiDER/gnss/processDelayFiles.py:173]: There was already an example command. I added another to cover the case of no gnss files.

**Comment by jlmaurer** [comment]
@dbekaert ok I changed the merge style to left. Not sure how to easily check how many GNSS stations are missing without some rather clunky code; related to outer merge but not sure how to only get the right side.

**Comment by dbekaert** [comment]
thanks @jlmaurer

**Pull request merged**